akka-cluster/src/main/scala/akka/cluster/protobuf/ClusterMessageSerializer.scala

```diff
@@ -336,7 +336,16 @@ class ClusterMessageSerializer(val system: ExtendedActorSystem) extends BaseSeri
 
     def memberFromProto(member: cm.Member) =
       new Member(addressMapping(member.getAddressIndex), member.getUpNumber, memberStatusFromInt(member.getStatus.getNumber),
-        member.getRolesIndexesList.asScala.map(roleMapping(_))(breakOut))
+        rolesFromProto(member.getRolesIndexesList.asScala))
+
+    @tailrec
+    def rolesFromProto(roleIndexes: Seq[Integer], resultSoFar: Set[String] = Set.empty): Set[String] = roleIndexes match {
+      case Seq() ⇒ resultSoFar + "team-default"
+      case Seq(head, tail @ _*) ⇒
+        val headrole = roleMapping(head)
+        if (headrole.startsWith("team-")) resultSoFar ++ roleIndexes.map(roleMapping(_)).toSet
+        else rolesFromProto(tail, resultSoFar + headrole)
+    }
```

Comment:
looks somewhat expensive. Not sure what collection type the `getRolesIndexesList.asScala` is. Perhaps we could iterate the raw java List and build the `Set` in a more efficient way than `++`?